src/chrome/content/rules/tomshardware.com.xml

```diff
@@ -0,0 +1,9 @@
+<!--
+  Certificate mismatch with https://tomshardware.com
+-->
+<ruleset name="tomshardware">
+	<target host="www.tomshardware.com" />
+	<test url="https://www.tomshardware.com/robots.txt" />
```

Comment:
You don't need a custom test URL like this

Reply:
Hmm. Yeah found that out too. Travis CI said I had to, but the error stayed even after I added it...
https://travis-ci.org/EFForg/https-everywhere/jobs/376267737#L595

Do you know by any chance how to fix this error?

Reply:
@bn4t Travis is right, the url in `<test url="https://www.tomshardware.com/robots.txt" />` isn't covered by the rule `<rule from="^http:"to="https:" />` since that url starts with an `https` and the rule covers `http`.

You can fix all of that by using this template,

```
<ruleset name="foo.com">
	<target host="foo.com" />
	<target host="www.foo.com" />

	<securecookie host=".+" name=".+" />

	<rule from="^http:" to="https:" />
</ruleset>
```

`tomshardware.com` also has some subdomains which you may try to find by using google search query `site:*.tomshardware.com -site:www.tomshardware.com` then iterate by adding any discovered subdomain `site:*.tomshardware.com -site:www.tomshardware.com -site:foo1.tomshardware.com` then `site:*.tomshardware.com -site:www.tomshardware.com -site:foo1.tomshardware.com -site:foo2.tomshardware.com` ...etc until you find them all, then add them to the ruleset

Reply:
Ah, ok. Thank you.
About the subdomains: haven't found any (except www) which have a correct certificate.

Reply:
Oh also since `https://tomshardware.com` uses an invalid certificate you'd have to add at the very beginning of the ruleset this comment

```
<!--
	Invalid Certificate:
		tomshardware.com
-->
```

and then add redirection rule 

```
	<securecookie host=".+" name=".+" />

	<rule from="^http://tomshardware\.com/" to="https://www.tomshardware.com/" />

	<rule from="^http:" to="https:" />
</ruleset>
```

Reply:
Thank you for helping me out so far :)
Unfortunately it still fails with the same error when validating: https://travis-ci.org/EFForg/https-everywhere/jobs/376316009

Any idea what it could be?